### DISCUSSION TITLE
GitHub Copilot Extensions Available in JetBrains IDEs

### DISCUSSION THREAD

**Discussion by D1M1TR10S** [community/community#146076] [Copilot Conversations]
![Copilot Extensions on JetBrains](https://github.com/user-attachments/assets/22fcb114-f262-42eb-a469-10fac538f493)

GitHub has released Copilot Extensions in public preview for the JetBrains IDE family. This change brings expanded capabilities to GitHub Copilot, allowing developers to leverage external tools, contexts and capabilities in their preferred JetBrains environment.

### What's New ✨
Copilot Extensions enable natural language interactions with external services and devtools without leaving your editor. Key additions from this update include:
- Full support across the JetBrains suite, including IntelliJ IDEA, PyCharm, and WebStorm
- Access to the complete GitHub Marketplace extensions ecosystem from JetBrains IDEs
- Seamless integration with external tools and services for JetBrains IDE users

### Getting started 🚀
- Update to the latest GitHub Copilot plugin for JetBrains
- Enable Copilot Extensions in IDE settings
- Install any desired extensions from [GitHub Marketplace](https://github.com/marketplace)
- Use an extension by typing '@' followed by the extension name and your prompt in the Copilot Chat window on your JetBrains IDE

To learn more, see our docs on [using](https://docs.github.com/en/copilot/using-github-copilot/asking-github-copilot-questions-in-your-ide#extending-copilot-chat-1) and [installing](https://docs.github.com/en/copilot/customizing-copilot/extending-the-capabilities-of-github-copilot-in-your-organization) Copilot Extensions.

### Building extensions for JetBrains IDEs
Developers can create custom extensions for internal use or publish them to the GitHub Marketplace. This flexibility allows teams to build specialized tools that fit their specific workflows, or drive discoverability for their product within the GitHub ecosystem. For more info, visit our [documentation on building Copilot Extensions](https://docs.github.com/en/copilot/building-copilot-extensions/about-building-copilot-extensions). Any extensions built as GitHub Apps will be available across all supported surfaces (JetBrains, VS Code, Visual Studio, GitHub.com, Mobile) and should be tested for all available surfaces.

### Requirements 📋
To use Copilot Extensions, you'll need:
- Active GitHub Copilot subscription
- Compatible JetBrains IDE
- Latest GitHub Copilot plugin installed on your IDE
- One or more Copilot Extensions installed

_Note that VS Code chat participants are not supported in this implementation._

### Feedback and questions 💬
We want to hear your feedback! Share your experience and suggestions for the GitHub product team, client teams (e.g. JetBrains-specific feedback) or third-party extension publishers in our [User Feedback Repository](https://github.com/copilot-extensions/user-feedback). Your input will help improve Copilot Extensions on JetBrains.

**Comment by rnavaneeth090902** [comment]
Any guide to build our own IntelliJ chat participants like VSC? @D1M1TR10S

**Comment by Delen-Luke-Beach** [comment]
I am unable to install the Github Copilot plugin in Webstorm. 

When I find the plugin from the marketplace within Webstorm and choose install it throws this error after hanging for a long time around 70%:
![image](https://github.com/user-attachments/assets/01564adb-8235-4fa0-8ecf-a9d316adff23)

Then, if I try to manually download the plugin from the Jetbrains website and use the 'install from zip' option in Webstorm, I get this:
![image](https://github.com/user-attachments/assets/26327259-c469-4884-a1c5-1381b832c1cc)

I can't even open the zip manually on Mac, not even with various CLI tools:
![image](https://github.com/user-attachments/assets/ccd1439f-d726-4ba6-879c-cd12b68b239f)

There is something wrong with the plugin, please can you upload a fixed version?

**Reply by Ravjhay** [reply]
gmzwins247.com

**Reply by Ravjhay** [reply]
please open the site

**Comment by JAIPUT** [comment]
[https://www.kasikornbank.com/th/personal/creditcard/pages/kbank-linepoints.aspx](gmt007)

**Comment by JAIPUT** [comment]
[https://www.kasikornbank.com/th/personal/creditcard/pages/kbank-linepoints.aspx](gm06t)

**Comment by mjomjo511** [comment]
> ![Copilot Extensions on JetBrains](https://github.com/user-attachments/assets/22fcb114-f262-42eb-a469-10fac538f493)
> 
> GitHub has released Copilot Extensions in public preview for the JetBrains IDE family. This change brings expanded capabilities to GitHub Copilot, allowing developers to leverage external tools, contexts and capabilities in their preferred JetBrains environment.
> 
> ### What's New ✨
> Copilot Extensions enable natural language interactions with external services and devtools without leaving your editor. Key additions from this update include:
> - Full support across the JetBrains suite, including IntelliJ IDEA, PyCharm, and WebStorm
> - Access to the complete GitHub Marketplace extensions ecosystem from JetBrains IDEs
> - Seamless integration with external tools and services for JetBrains IDE users
> 
> ### Getting started 🚀
> - Update to the latest GitHub Copilot plugin for JetBrains
> - Enable Copilot Extensions in IDE settings
> - Install any desired extensions from [GitHub Marketplace](https://github.com/marketplace)
> - Use an extension by typing '@' followed by the extension name and your prompt in the Copilot Chat window on your JetBrains IDE
> 
> To learn more, see our docs on [using](https://docs.github.com/en/copilot/using-github-copilot/asking-github-copilot-questions-in-your-ide#extending-copilot-chat-1) and [installing](https://docs.github.com/en/copilot/customizing-copilot/extending-the-capabilities-of-github-copilot-in-your-organization) Copilot Extensions.
> 
> ### Building extensions for JetBrains IDEs
> Developers can create custom extensions for internal use or publish them to the GitHub Marketplace. This flexibility allows teams to build specialized tools that fit their specific workflows, or drive discoverability for their product within the GitHub ecosystem. For more info, visit our [documentation on building Copilot Extensions](https://docs.github.com/en/copilot/building-copilot-extensions/about-building-copilot-extensions). Any extensions built as GitHub Apps will be available across all supported surfaces (JetBrains, VS Code, Visual Studio, GitHub.com, Mobile) and should be tested for all available surfaces.
> 
> ### Requirements 📋
> To use Copilot Extensions, you'll need:
> - Active GitHub Copilot subscription
> - Compatible JetBrains IDE
> - Latest GitHub Copilot plugin installed on your IDE
> - One or more Copilot Extensions installed
> 
> _Note that VS Code chat participants are not supported in this implementation._
> 
> ### Feedback and questions 💬
> We want to hear your feedback! Share your experience and suggestions for the GitHub product team, client teams (e.g. JetBrains-specific feedback) or third-party extension publishers in our [User Feedback Repository](https://github.com/copilot-extensions/user-feedback). Your input will help improve Copilot Extensions on JetBrains.

**Comment by 01001989146** [comment]
01001989146

**Comment by 01001989146** [comment]
shryfalshamy.1985eee@gmail.com

**Comment by 01001989146** [comment]
شريف على محمد الشامى العنوان جمهوريه مصر العربيه البحيره دمنهور قريه الابعديه الرقم القومي ٢٨٥٠١٠٣١٨٠٠٤٣٣رقم٠١٠٠١٩٨٩١٤٦ حلم

**Comment by 01001989146** [comment]
[شريفعلىمحمدالشامىالعنوانجمهوريهمصرالعربيهالبحيرهدمنهورقريهالابعديهالرقمالقومي٢٨٥٠١٠٣١٨٠٠٤٣٣رقم٠١٠٠١٩٨٩١٤٦حلم](url)

**Comment by ken-kuro** [comment]
Anyone got it working? I'm on the latest version of Jetbrains IDE with the latest version of plugin as well
